Extension/src/LanguageServer/ui.ts

```diff
@@ -426,14 +433,21 @@ export class LanguageStatusUI {
     private showConfigureIntelliSenseButton: boolean = false;
     private configureIntelliSenseTimeout?: NodeJS.Timeout;
 
-    public async ShowConfigureIntelliSenseButton(show: boolean, client?: Client): Promise<void> {
-        if (!await telemetry.showStatusBarIntelliSenseButton() || client !== this.currentClient) {
-            return;
-        }
+    public async ShowConfigureIntelliSenseButton(show: boolean, client?: Client, configurationType?: ConfigurationType, sender?: string): Promise<void> {
         this.showConfigureIntelliSenseButton = show;
```

Comment:
I don't think this is the correct change -- you're causing setShowConfigureIntelliSenseButton to be run when without checking `!await telemetry.showStatusBarIntelliSenseButton() || client !== this.currentClient`

Reply:
i.e. I think the code that was moved up should be moved back down and the check for `client !== this.currentClient` should be moved up.

Reply:
I fixed it.